src/c++/perf_analyzer/concurrency_manager.cc

```diff
@@ -91,7 +91,18 @@ cb::Error
 ConcurrencyManager::ChangeConcurrencyLevel(
     const size_t concurrent_request_count)
 {
-  if (on_sequence_model_ && async_) {
+  PauseSequenceWorkers();
+  ReconfigThreads(concurrent_request_count);
+  ResumeSequenceWorkers();
+
+  std::cout << "Request concurrency: " << concurrent_request_count << std::endl;
+  return cb::Error::Success;
+}
+
+void
+ConcurrencyManager::PauseSequenceWorkers()
+{
+  if (on_sequence_model_) {
```

Comment:
Can you explain again why it's only sequence models that need to be paused while the concurrency is changed?

Reply:
Sequence models are the only ones that need to "finish" anything to make sure we have a fresh start for the next rate.